examples/autoignition-mpi.py

```diff
@@ -222,39 +222,70 @@ def my_rhs(t, state):
                                boundaries=boundaries, eos=eos)
                 + eos.get_species_source_terms(state))
 
-    def my_checkpoint(step, t, dt, state):
-        reaction_rates = eos.get_production_rates(state)
-        viz_fields = [("reaction_rates", reaction_rates)]
-        return sim_checkpoint(discr, visualizer, eos, cv=state,
-                              vizname=casename, step=step,
-                              t=t, dt=dt, nstatus=nstatus, nviz=nviz,
-                              constant_cfl=constant_cfl, comm=comm,
-                              viz_fields=viz_fields)
-
-    try:
-        (current_step, current_t, current_state) = \
-            advance_state(rhs=my_rhs, timestepper=timestepper,
-                checkpoint=my_checkpoint,
-                get_timestep=get_timestep, state=current_state,
-                t=current_t, t_final=t_final)
-    except ExactSolutionMismatch as ex:
-        error_state = True
-        current_step = ex.step
-        current_t = ex.t
-        current_state = ex.state
+    def my_checkpoint(step, t, dt, state, force=False):
+        from mirgecom.simutil import check_step
+        do_status = force or check_step(step=step, interval=nstatus)
+        do_viz = force or check_step(step=step, interval=nviz)
+        do_health = force or check_step(step=step, interval=nhealth)
+
+        if do_status or do_viz or do_health:
+            dv = eos.dependent_vars(state)
+            reaction_rates = eos.get_production_rates(state)
+
+        if do_status:  # This is bad, logging already completely replaces this
```

Comment:
> This is bad, logging already completely replaces this

Then why have it? (As opposed to taking this opportunity to kill it?)

Reply:
I'd love to.  It seems we have not learned yet how to effectively add user-defined quantities to logging, and at present no way to control its frequency.  I'll file an issue about that, but will you please have another look at the `autoignition-mpi.py` and `vortex-mpi.py` when you get a chance?  In (100df77), those examples have an example of what it would look like to use logging instead of the un-customizable built-in status message utility.

Reply:
kilt